contrib/inventory/ec2.py

```diff
@@ -789,11 +793,18 @@ def add_instance(self, instance, region):
             else:
                 hostname = getattr(instance, self.hostname_variable)
 
+        # set the hostname from route53
+        if self.route53_enabled and self.route53_hostnames:
+            route53_names = self.get_instance_route53_names(instance)
+            for name in route53_names:
+                if name.endswith(self.route53_hostnames):
+                    hostname = name
+
         # If we can't get a nice hostname, use the destination address
         if not hostname:
             hostname = dest
         else:
-            hostname = self.to_safe(hostname).lower()
+            hostname = hostname.lower()
```

Comment:
Removing the to_safe parameter here will affect the output whether or not the route53_hostnames option is enabled, breaking any playbook that uses the existing hostnames. I understand that route53 names can't be filtered through to_safe since it strips DNS-valid characters, so if you add a conditional then this would be ok.